canopen/objectdictionary/datatypes_struct.py

```diff
@@ -0,0 +1,62 @@
+import struct
+
+class UnsignedN:
+    """ struct-like class for packing and unpacking unsigned integers of arbitrary width.
+    The width must be a multiple of 8 and must be between 8 and 64.
+    """
+    def __init__(self, width: int):
+        self.width = width
+        if width % 8 != 0:
+            raise ValueError("Width must be a multiple of 8")
+        if width <= 0 or width > 64:
+            raise ValueError("Invalid width for UnsignedN")
+        elif width <= 8:
+            self.__st = struct.Struct("B")
+        elif width <= 16:
+            self.__st = struct.Struct("<H")
+        elif width <= 32:
+            self.__st = struct.Struct("<L")
+        elif width <= 64:
```

Comment:
I would prefer `else` here, makes it easier to reason about `__st` always being initialized in an instance.